engine/pkg/externalresource/manager/gc_coordinator.go

```diff
@@ -175,68 +172,59 @@ func (c *DefaultGCCoordinator) gcByStatusSnapshots(
 	}
 
 	var (
-		toGC []engineModel.JobID
-
-		// toRemove is used to remove meta records when
-		// the associated executors are offline.
-		// TODO adjust the mechanism when we implement S3 support.
-		toRemove []model.ExecutorID
+		toGCJobs      []engineModel.JobID
+		toGCExecutors []model.ExecutorID
 	)
 	for _, resMeta := range resources {
 		if _, exists := jobSnapshot[resMeta.Job]; !exists {
 			// The resource belongs to a deleted job.
-			toGC = append(toGC, resMeta.Job)
+			toGCJobs = append(toGCJobs, resMeta.Job)
 			continue
 		}
 
 		if _, exists := executorSet[resMeta.Executor]; !exists {
 			// The resource belongs to an offlined executor.
-			toRemove = append(toRemove, resMeta.Executor)
+			tp, resName, err := resModel.PasreResourceID(resMeta.ID)
+			if err != nil {
+				return err
+			}
+			if tp == resModel.ResourceTypeLocalFile ||
+				tp == resModel.ResourceTypeS3 && resName == s3.DummyResourceName {
+				toGCExecutors = append(toGCExecutors, resMeta.Executor)
```

Comment:
Do multiple resources reference the same executor?

Reply:
fix it, PTAL!